drivers/platform/linux/axi_io.c

```diff
@@ -0,0 +1,128 @@
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/mman.h>
+#include "error.h"
+#include "axi_io.h"
+
+/**
+ * @brief AXI IO through UIO read function.
+ * @param base - UIO index (/dev/uioX).
+ * @param offset - Address offset.
+ * @param data - Location where read data will be stored.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t axi_io_read(uint32_t base, uint32_t offset, uint32_t *data)
+{
+	char buf[16];
+	int ret;
+	int uio_fd;
+	void *uio_addr;
+
+	sprintf(buf, "/dev/uio%d", base);
+
+	uio_fd = open(buf, O_RDWR);
+	if (uio_fd < 0) {
+		printf("%s: Can't open %s\n\r", __func__, buf);
+		return FAILURE;
+	}
+
+	uio_addr = mmap(NULL,
+			0x1000,
+			PROT_READ|PROT_WRITE,
+			MAP_SHARED,
+			uio_fd,
+			0);
+
+	*data = (*((unsigned *) (uio_addr + offset)));
+
+	ret = close(uio_fd);
+	if (ret < 0) {
+		printf("%s: Can't close %s\n\r", __func__, buf);
+		return FAILURE;
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief AXI IO through UIO specific write function.
+ * @param base - UIO index (/dev/uioX).
+ * @param offset - Address offset.
+ * @param data - Data to be written.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t axi_io_write(uint32_t base, uint32_t offset, uint32_t data)
+{
+	char buf[16];
+	int ret;
+	int uio_fd;
+	void *uio_addr;
+
+	sprintf(buf, "/dev/uio%d", base);
+
+	uio_fd = open(buf, O_RDWR);
+	if (uio_fd < 0) {
+		printf("%s: Can't open %s\n\r", __func__, buf);
+		return FAILURE;
+	}
+
+	uio_addr = mmap(NULL,
+			0x1000,
+			PROT_READ|PROT_WRITE,
+			MAP_SHARED,
+			uio_fd,
+			0);
+
+	*((unsigned *) (uio_addr + offset)) = data;
```

Comment:
Changing this to
`*((uint32_t *)uio_addr + offset) = data;`
should solve the Codancy issue.